### DISCUSSION TITLE
Livewire not updating html after download

### DISCUSSION THREAD

**Discussion by fabio-ivona** [livewire/livewire#4057] [1. Help] [answered]
Just wondering why livewire calls `skipRender()` when downloading a file

```php
//SupportFileDownloads.php

Livewire::listen('action.returned', function ($component, $action, $returned) {
    if ($this->valueIsntAFileResponse($returned)) return;

    // do download stuff

    $component->skipRender();  // <-- why??
});

this way I cannot update the UI after a download to (maybe) hiding the download button


is there any reason why rendering is disabled? anyone can help me to understand why?

**Comment by joshhanley** [comment, marked as answer]
@fabio-ivona not sure exactly why that is in there. 

I suspect it is due to Livewire trying to simulate how a normal file download happens. 

Like if you hit a Laravel controller endpoint that downloads a file, that is all that endpoint does. Where as Livewire triggers the download using javascript instead of returning a download response like laravel, and so maybe this was to simulate that.

I don't know what the implications would be for removing it, but PR's are welcome 🙂

Hope this helps!

**Comment by supergreentony** [comment]
Hi,

I have the same problem and I wonder how can I pass over it?

Here is my function :
```php
public function download()
{
    //The "is_downloaded" attribute is updated for all products in selection
    Product::whereIn('id', $this->selection)->update(['is_downloaded' => true]);

    return downloadable()->downloadFiles($this->selection, Product::class);
}
```
I need the view to be updated with the new data.
Is it possible?

Thank you!

**Reply by fabio-ivona** [reply]
It has been fixed with this PR

https://github.com/livewire/livewire/pull/4191

But probably you should refresh the products already loaded in the component

**Reply by supergreentony** [reply]
Oh nice !
I didn't see this PR. I will update livewire.

Thank you :)
